src/m365/teams/commands/chat/chat-get.ts

```diff
@@ -0,0 +1,200 @@
+import { AadUserConversationMember, Chat, ConversationMember } from '@microsoft/microsoft-graph-types';
+import { AxiosRequestConfig } from 'axios';
+import Auth from '../../../../Auth';
+import * as os from 'os';
+import request from '../../../../request';
+import { Logger } from '../../../../cli';
+import {
+  CommandOption
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+import { validation } from '../../../../utils/validation';
+import { accessToken } from '../../../../utils/accessToken';
+import { ODataResponse } from '../../../../utils/odata';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  id?: string;
+  participants?: string;
+  name?: string;
+}
+
+class TeamsChatGetCommand extends GraphCommand {
+  public get name(): string {
+    return commands.CHAT_GET;
+  }
+
+  public get description(): string {
+    return 'Get a chat conversations by id, participants or chat name';
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs, cb: () => void): Promise<void> {
```

Comment:
`commandAction` is treated as a sync method in the CLI runtime. Won't making it async cause issues?

Reply:
Good question. Because it is marked `async`, the `commandAction` function will now return a `promise` instead of `void`. I have not seen any issues with the execution. Of course the cli is not awaiting the promise, but that's okay as we're using a callback function anyway. 

Strictly speaking though, I can understand if you would like it better to adhere to the interface of commandAction returning void. Everyone else is doing it as well.

I've refactored it to return void. I should refactor the teams chat message send command for this as well. Let me know how you want to organise a refactoring of an existing command.

Reply:
We can use async for other methods that we call inside `commandAction`, as long as the main flow is sync and calls the callback to signal command execution completed to the CLI runtime.

Reply:
https://github.com/pnp/cli-microsoft365/issues/3106